### PR TITLE
[Bug] fix scala.collection.JavaConversions to scala.collection.JavaConverter

### PR DESCRIPTION
**Pull request by GuoPhilipse**
### What is the purpose of the change

fix  scala.collection.JavaConversions._  —》  scala.collection.JavaConverters._

### Related issues/PRs

Related issues: #3994
Related pr:#3998

### Brief change log

fix  scala.collection.JavaConversions._  —》  scala.collection.JavaConverters._

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/ParserVarLabelInterceptor.scala [hunk 1]
```diff
@@ -26,15 +26,17 @@ import org.apache.linkis.protocol.utils.TaskUtils
...
   override def apply(jobRequest: JobRequest, logAppender: lang.StringBuilder): JobRequest = {
     jobRequest match {
       case requestPersistTask: JobRequest =>
         val variableMap =
-          TaskUtils.getVariableMap(requestPersistTask.getParams.asInstanceOf[util.Map[String, Any]])
+          TaskUtils
+            .getVariableMap(requestPersistTask.getParams.asInstanceOf[util.Map[String, Any]])
+            .asScala
         val labels = requestPersistTask.getLabels
         if (variableMap.contains(LabelKeyConstant.TENANT_KEY)) {
```
Comment (peacewong) [inline linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/ParserVarLabelInterceptor.scala:41]: can use containsKey

File: linkis-engineconn-plugins/jdbc/src/main/scala/org/apache/linkis/manager/engineplugin/jdbc/executor/JDBCEngineConnExecutor.scala [hunk 2]
```diff
@@ -261,7 +261,9 @@ class JDBCEngineConnExecutor(override val outputPrintLimit: Int, val id: Int)
       val md = resultSet.getMetaData
       val metaArrayBuffer = new ArrayBuffer[(String, String)]()
       for (i <- 1 to md.getColumnCount) {
-        metaArrayBuffer.add(Tuple2(md.getColumnName(i), JDBCHelper.getTypeStr(md.getColumnType(i))))
+        metaArrayBuffer.asJava.add(
```
Comment (peacewong) [inline linkis-engineconn-plugins/jdbc/src/main/scala/org/apache/linkis/manager/engineplugin/jdbc/executor/JDBCEngineConnExecutor.scala:264]: can use append

File: linkis-public-enhancements/linkis-jobhistory/src/main/scala/org/apache/linkis/jobhistory/service/impl/JobHistoryQueryServiceImpl.scala [hunk 3]
```diff
@@ -238,7 +237,7 @@ class JobHistoryQueryServiceImpl extends JobHistoryQueryService with Logging {
   }
 
   /* private def queryTaskList2RequestPersistTaskList(queryTask: java.util.List[QueryTask]): java.util.List[RequestPersistTask] = {
-    import scala.collection.JavaConversions._
+    import scala.collection.JavaConverters._
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-jobhistory/src/main/scala/org/apache/linkis/jobhistory/service/impl/JobHistoryQueryServiceImpl.scala:240]: can remove

File: linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala [hunk 4]
```diff
@@ -31,7 +31,7 @@ import org.apache.commons.lang3.StringUtils
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
-import scala.collection.JavaConversions._
```
Comment (peacewong) [inline linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala:34]: plugins dir can delete

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/UJESSQLDatabaseMetaData.scala [hunk 5]
```diff
@@ -328,7 +328,7 @@ class UJESSQLDatabaseMetaData(ujesSQLConnection: UJESSQLConnection) extends Data
     val getDBSAction = GetDBSAction.builder().setUser(getUserName).build()
     val dBSResult = ujesSQLConnection.ujesClient.getDBS(getDBSAction)
     val dbsName = dBSResult.getDBSName()
```
[the remark this answers is not in the document]
Comment (GuoPhilipse) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/UJESSQLDatabaseMetaData.scala:328]: Thanks @peacewong.  other comments  have been addressed except this one, not quite clear about this comment ,could you please explain more about this one ?

[the remark this answers is not in the document]
Comment (peacewong) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/UJESSQLDatabaseMetaData.scala:328]: Hello, you need to compile the linkis-jdbc-driver module separately. I don’t know if there is a problem with this modification.

[the remark this answers is not in the document]
Comment (GuoPhilipse) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/UJESSQLDatabaseMetaData.scala:328]: I saw
 ```
<module>linkis-jdbc-driver</module>
```
was comment out as
limage

If we need to confirm the compile, maybe we can let it  join the compile.  i don't know whether sth happed for this before.

[the remark this answers is not in the document]
Comment (GuoPhilipse) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/UJESSQLDatabaseMetaData.scala:420]: hello @peacewong  i have rechecked the code and recompiled the jdbc module successfully,

**Review by peacewong — DISMISSED** [review]
LGTM.

File: linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala (hunk 4 above)
Comment (GuoPhilipse) [reply linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala:34]: code updated~

File: linkis-public-enhancements/linkis-jobhistory/src/main/scala/org/apache/linkis/jobhistory/service/impl/JobHistoryQueryServiceImpl.scala (hunk 3 above)
Comment (GuoPhilipse) [reply linkis-public-enhancements/linkis-jobhistory/src/main/scala/org/apache/linkis/jobhistory/service/impl/JobHistoryQueryServiceImpl.scala]: done

**Comment by GuoPhilipse** [comment]
ERROR: 1 dead links found!
[✖] https://linkis.apache.org/zh-CN/community/development-specification/commit-message → Status: 404

[✖] https://linkis.apache.org/community/development-specification/commit-message → Status: 404

cc @peacewong @casionone

**Comment by GuoPhilipse** [comment]
> ERROR: 1 dead links found! [✖] https://linkis.apache.org/zh-CN/community/development-specification/commit-message → Status: 404
> 
> [✖] https://linkis.apache.org/community/development-specification/commit-message → Status: 404
> 
> cc @peacewong @casionone

I have just try to fix it in #4041

**Comment by peacewong** [comment]
> > ERROR: 1 dead links found! [✖] https://linkis.apache.org/zh-CN/community/development-specification/commit-message → Status: 404
> > [✖] https://linkis.apache.org/community/development-specification/commit-message → Status: 404
> > cc @peacewong @casionone
> 
> I have just try to fix it in #4041

ok

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**